File: src/bubbleSort.js
/**
 * 冒泡排序
 * @param {Array} arr 待排序数组
 */
function bubbleSort(arr) {
  const len = arr.length;

  for (let i = 0; i < len; i++) {
    let flag = false;
    for (let j = 0; j < len - i - 1; j++) {
      if (arr[j] > arr[j + 1]) {
        const temp = arr[j];
        arr[j] = arr[j + 1];
        arr[j + 1] = temp;
        flag = true;
      }
    }
    if (!flag) break;
  }
  return arr;
}

module.exports = bubbleSort;

File: java/src/BubbleSort.java
public class BubbleSort {

    public static void main(String[] args) {
        int[] arr = new int[]{6, 4, 3, 8, 1};
        int[] newArr = new BubbleSort().bubbleSort(arr);
        System.out.println(newArr);
    }

    public int[] bubbleSort(int[] arr) {
        int len = arr.length;

        for (int i = 0; i < len; i++) {
            for (int j = 0; j < len - i - 1; j++) {
                if (arr[j] > arr[j + 1]) {
                    int tmp = arr[j];
                    arr[j] = arr[j + 1];
                    arr[j + 1] = tmp;
                }
            }
        }

        return arr;
    }
}
